src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -210,39 +231,43 @@ public override Task Save(DatabaseViewContext context, DatabaseInfo obj)
         public override Task<string> Script(DatabaseViewContext context, DatabaseInfo obj)
         {
             var script = ConfigureDatabase(
-                context.Parameters.ConnectionUri,
+                context.Parameters,
                 obj,
                 context.Parameters.IsNewObject ? ConfigAction.Create : ConfigAction.Update,
                 RunType.ScriptToWindow);
             return Task.FromResult(script);
         }
 
-        private CDataContainer CreateDatabaseDataContainer(string connectionUri, ConfigAction configAction, DatabaseInfo? database = null)
+        private CDataContainer CreateDatabaseDataContainer(InitializeViewRequestParams requestParams, ConfigAction configAction, DatabaseInfo? database = null)
         {
-            ConnectionInfo connectionInfo = this.GetConnectionInfo(connectionUri);
+            ConnectionInfo connectionInfo = this.GetConnectionInfo(requestParams.ConnectionUri);
             CDataContainer dataContainer = CDataContainer.CreateDataContainer(connectionInfo, databaseExists: configAction != ConfigAction.Create);
             if (dataContainer.Server == null)
             {
                 throw new InvalidOperationException(serverNotExistsError);
             }
-            string objectUrn = (configAction != ConfigAction.Create && database != null)
-                ? string.Format(System.Globalization.CultureInfo.InvariantCulture,
-                    "Server/Database[@Name='{0}']",
-                    Urn.EscapeString(database.Name))
-                : string.Format(System.Globalization.CultureInfo.InvariantCulture,
-                    "Server");
+            string objectUrn = requestParams.ObjectUrn;
+            if (requestParams.IsNewObject)
```

Comment:
Why is this check being added? Do you not need an object URN for existing objects as well?

Reply:
Ideally you should be able to use the existing case as is, since it includes the database name if it was provided. Was it not working for the properties dialog?

Reply:
No, ConfigAction.Create is stopping to use the case, objectUrn should be "Server[@Name='SQLTools2022-3']/Database[@Name='keep_AdventureWorksDW']" but we get "Server" instead, I can make modifications and use it

Reply:
Passing configAction(Update) and DatabaseInfo works fine to use the existing case to get the smo object. reverted the change too! Thanks